src/v0/destinations/braze/README.md

```diff
@@ -0,0 +1,240 @@
+# Braze Destination
+
+Implementation in **Javascript**
+
+## Configuration
+
+### Required Settings
+
+- **REST API Key**: Required for authentication with Braze REST API
+
+  - Must have appropriate permissions for the operations you want to perform
+  - For user deletion, the API key must have the `users.delete` permission
+
+- **Data Center**: Specifies the Braze data center to use
+
+  - Format: `[REGION]-[NUMBER]` (e.g., `US-01`, `EU-01`)
+  - Available data centers:
+    - US: `US-01` through `US-08` (e.g., `US-01`, `US-02`, etc.)
+    - EU: `EU-01` through `EU-02` (e.g., `EU-01`, `EU-02`)
+    - Other regions: Consult Braze documentation for the latest available regions
+
+- **App Key**: Required for SDK implementations (device mode)
+
+### Optional Settings
+
+- **Support Deduplication**: Enable to avoid sending duplicate user attributes
+- **Prefix Properties**: Add a prefix to all properties sent to Braze
+- **Enable Nested Array Operations**: Enable operations on nested arrays
+- **Enable Nested Object Operations**: Enable operations on nested objects
+- **Track Anonymous User**: Enable tracking of users without a userId
+- **Enable Subscription Group in Group Call**: Enable subscription group functionality in group calls
+
+## Integration Functionalities
+
+### Supported Message Types
+
+- Identify
+- Track
+- Page
+- Screen
+- Group
+- Alias
+
+### Batching Support
+
+- **Supported**: Yes
+- **Message Types**: All message types
+- **Batch Limits**:
+  - Track events (includes Page and Screen events): 75 events per batch
+  - Identify events: 50 events per batch
+  - Alias events: 50 events per batch
+  - Subscription events: 25 events per batch
+  - User deletion: 50 users per batch
+
+### Intermediate Calls
+
+- **Supported**: Yes
+- **Use Case**: Identity resolution through the Braze Identify API
+- **Endpoint**: `/users/identify`
+- This functionality merges anonymous users (with anonymousId) with identified users (with userId/external_id)
+
+```Javascript
+// The condition that leads to intermediate identify call:
+const brazeExternalID = getDestinationExternalID(message, 'brazeExternalId') || message.userId;
+if ((message.anonymousId || isAliasPresent) && brazeExternalID) {
+  await processIdentify({ message, destination });
+} else {
+  collectStatsForAliasMissConfigurations(destination.ID);
+}
+```
+
+### Merging Behavior
+
+- **User Identity Resolution**:
```

Comment:
Before this section lets add a talking point that when an alias only user gets created - example if only anonymous identify or track call is made we will create an alias only profile
where `alias_label: rudder_id , alias_name: anon_id_value`

And in case rudderstack external_id or user_id present we also add it as `braze external_id`